src/main/HttpHandler.ts

```diff
@@ -0,0 +1,9 @@
+import { HttpContext } from './HttpContext.js';
+
+export type HttpNext = () => Promise<void>;
+
+export type HttpHandlerFn = (ctx: HttpContext, next: HttpNext) => Promise<void>;
```

Comment:
Why do we need this, if the function `handle` already has `(ctx: HttpContext, next: HttpNext)` as well?

Reply:
Yep, I was also frowning at this, a little. Initially I made it `handle: HttpHandlerFn` in the `HttpHandler` interface. However, VS Code autocomplete no longer treats it as a method (kinda rightfully so) and offers you to create a regular field with an arrow function instead, which is 100% not how the API is intended to work.

So essentially there's still semantic difference between the two — even if both work in practice.

Given that I'm also using this type in other places (`HttpChain`, `compose`, etc) I decided to just keep both of them.

Lmk if you have a better solution though ✨

Reply:
That's interesting about VS Code not considering it a method 🤔. And there's no way to disable this?

Reply:
No, it's not VS Code -specific I don't think. It's TypeScript, the following declarations are not 100% equivalent, even though it's very likely that an object that satisfies one of them would satisfy another.

```
interface One<A, B> {
    foo(arg: A): B;
}

interface Two<A, B> {
    foo: (arg: A) => B;
}
```

I _think_ there might be an exception though, because `One` might have an implicit `this: One`, whereas in `Two` this wouldn't be the case.